test/e2e/proxy_test.go

```diff
@@ -95,47 +109,84 @@ func skipIfKubeProxyEnabled(t *testing.T, data *TestData) {
-	return hostname, err
-}
-
-func probeClientIPFromNode(node string, baseUrl string) (string, error) {
-	url := fmt.Sprintf("%s/%s", baseUrl, "clientip")
-	_, hostPort, _, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", url))
+func probeClientIPEndpointFromNode(node string, url string) (string, error) {
+	stdout, err := probeEndpointFromNode(node, url, clientIPEp)
 	if err != nil {
 		return "", err
 	}
-	host, _, err := net.SplitHostPort(hostPort)
+	host, _, err := net.SplitHostPort(stdout)
 	return host, err
 }
 
-func probeFromPod(data *TestData, pod string, url string) error {
-	_, _, err := data.runCommandFromPod(testNamespace, pod, busyboxContainerName, []string{"wget", "-O", "-", url, "-T", "1"})
-	return err
+func probeEndpointFromNode(node string, baseURL string, endpoint agnhostEndpoint) (string, error) {
+	url := fmt.Sprintf("%s/%s", baseURL, endpoint)
+	rc, stdout, stderr, err := RunCommandOnNode(node, fmt.Sprintf("curl --connect-timeout 5 --retry 5 --retry-connrefused %s", url))
+	if err != nil {
+		return "", fmt.Errorf("rc: %d, stdout: %s, stderr: %s, err: %v", rc, stdout, stderr, err)
+	}
+	return stdout, nil
 }
 
-func probeHostnameFromPod(data *TestData, pod string, baseUrl string) (string, error) {
-	url := fmt.Sprintf("%s/%s", baseUrl, "hostname")
-	hostname, _, err := data.runCommandFromPod(testNamespace, pod, busyboxContainerName, []string{"wget", "-O", "-", url, "-T", "1"})
-	return hostname, err
+func probeFromPod(data *TestData, pod string, os string, url string) error {
+	_, err := probeEndpointFromPod(data, pod, os, url, emptyEp)
+	return err
 }
 
-func probeClientIPFromPod(data *TestData, pod string, baseUrl string) (string, error) {
-	url := fmt.Sprintf("%s/%s", baseUrl, "clientip")
-	hostPort, _, err := data.runCommandFromPod(testNamespace, pod, busyboxContainerName, []string{"wget", "-O", "-", url, "-T", "1"})
+func probeClientIPEndpointFromPod(data *TestData, pod string, os string, url string) (string, error) {
+	stdout, err := probeEndpointFromPod(data, pod, os, url, clientIPEp)
 	if err != nil {
 		return "", err
 	}
-	host, _, err := net.SplitHostPort(hostPort)
+	host, _, err := net.SplitHostPort(stdout)
 	return host, err
 }
 
+func probeEndpointFromPod(data *TestData, pod string, os string, baseUrl string, endpoint agnhostEndpoint) (string, error) {
+	url := fmt.Sprintf("%s/%s", baseUrl, endpoint)
+	cmd := []string{"wget", "-O", "-", url, "-T", "5"}
+	if os == "windows" {
+		cmd = []string{"curl", "--connect-timeout", "5", "--retry", "5", url}
+	}
+	stdout, stderr, err := data.runCommandFromPod(testNamespace, pod, agnhostContainerName, cmd)
+	if err != nil {
+		return "", fmt.Errorf("stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+	}
+	return stdout, nil
+}
+
+func pickNodes(isIPv6 bool) ([]string, []string, corev1.IPFamily) {
```

Comment:
It's a little strange to make a function named `pickNodes` to return the ip family when the argument is already `isIPv6`, the caller itself should have the information. And there should be no difference if keeping L164, L165 and L167 on caller side.

Reply:
Done